File: backend/internal/model/podcast.go
package model

import (
	"time"

	"github.com/google/uuid"
)

// Podcast はポッドキャスト番組を表すドメインモデルです。
type Podcast struct {
	ID          uuid.UUID `db:"id" json:"id"`
	ItunesID    *int64    `db:"itunes_id" json:"itunes_id,omitempty"`
	Title       string    `db:"title" json:"title"`
	Author      *string   `db:"author" json:"author,omitempty"`
	Description *string   `db:"description" json:"description,omitempty"`
	FeedURL     *string   `db:"feed_url" json:"feed_url,omitempty"`
	ArtworkURL  *string   `db:"artwork_url" json:"artwork_url,omitempty"`
	ItunesURL   *string   `db:"itunes_url" json:"itunes_url,omitempty"`
	Genre       *string   `db:"genre" json:"genre,omitempty"`
	SourceType  string    `db:"source_type" json:"source_type"`
	SourceURL   *string   `db:"source_url" json:"source_url,omitempty"`
	CreatedAt   time.Time `db:"created_at" json:"created_at"`
	UpdatedAt   time.Time `db:"updated_at" json:"updated_at"`
}

File: frontend/src/lib/schemas/podcast.ts
import { z } from "zod";
import { uuidSchema, datetimeSchema } from "./common";

/** Podcast スキーマ（API レスポンス） */
export const podcastSchema = z.object({
  id: uuidSchema,
  itunes_id: z.number().nullish(),
  title: z.string(),
  author: z.string().nullish(),
  description: z.string().nullish(),
  feed_url: z.string().nullish(),
  artwork_url: z.string().nullish(),
  itunes_url: z.string().nullish(),
  genre: z.string().nullish(),
  source_type: z.enum(["itunes", "radiko", "manual"]),
  source_url: z.string().nullish(),
  created_at: datetimeSchema,
  updated_at: datetimeSchema,
});

export type Podcast = z.infer<typeof podcastSchema>;

/** 検索結果の番組アイテム */
export const podcastSearchItemSchema = z.object({
  id: uuidSchema,
  title: z.string(),
  author: z.string().nullish(),
  artwork_url: z.string().nullish(),
  average_rating: z.number(),
  total_ratings: z.number(),
});

export type PodcastSearchItem = z.infer<typeof podcastSearchItemSchema>;

/** 番組検索結果 */
export const podcastSearchResultSchema = z.object({
  podcasts: z.array(podcastSearchItemSchema),
  total: z.number(),
});

export type PodcastSearchResult = z.infer<typeof podcastSearchResultSchema>;
